python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+
+                # # Apply gradient along Y axis to the image
+                # # and parallel overscan regions.
+                self.amplifierAddYGradient(ampData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+                self.amplifierAddYGradient(parallelOscanData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+
+        # 6. Add Parallel overscan xtalk.
+        # TODO: DM-43286
+
+            if self.config.doAddSerialOverscan:
+                # 1. We grow the image to the parallel oversan region
+                # (we do this instead of using the whole raw region
+                # in case there are prescan regions)
+                grownImageBBox = imageBBox.expandedTo(parallelOscanBBox)
+                # 2. Now we grow the serial overscan region
+                # to include the corners
+                serialOscanBBox = geom.Box2I(
+                    geom.Point2I(serialOscanBBox.getMinX(),
+                                 grownImageBBox.getMinY()),
+                    geom.Extent2I(serialOscanBBox.getWidth(),
+                                  grownImageBBox.getHeight()),
+                )
+                serialOscanData = exposure.image[serialOscanBBox]
+
+                # 7. Add bias level to the whole region of amplifiers in ADU
+                self.amplifierAddNoise(ampData, self.config.biasLevel,
+                                       self.config.readNoise / self.config.gain)
+                self.amplifierAddNoise(parallelOscanData, self.config.biasLevel,
+                                       self.config.readNoise / self.config.gain)
+                self.amplifierAddNoise(serialOscanData, self.config.biasLevel,
+                                       self.config.readNoise / self.config.gain)
+
+                # 8. Apply serial overscan in ADU
+                # Apply gradient along X axis to the whole amp
+                # First the serial overscan and corners region
+                self.amplifierAddXGradient(serialOscanData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+                # And second the image and parallel overscan region
+                self.amplifierAddXGradient(ampData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+                self.amplifierAddXGradient(parallelOscanData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+
+        if self.config.doGenerateAmpDict:
+            expDict = dict()
+            for amp in exposure.getDetector():
+                expDict[amp.getName()] = exposure
+            return expDict
+        else:
+            return exposure
+
+    def addGain(self, ampData, gain):
+        ampArr = ampData.array
```

Comment:
Docstring here, indicating that despite the name, this is a multiplicative effect (I chose bad names when I wrote the original class).

Reply:
Done and changed the name to applyGain.